lib/cyrusdb_sql.c

```diff
@@ -649,8 +665,16 @@ static int fetch_cb(void *rock,
     struct fetch_rock *frock = (struct fetch_rock *) rock;
 
     if (frock->data) {
-        *(frock->data) = xrealloc(*(frock->data), datalen);
-        memcpy(*(frock->data), data, datalen);
+        if (datalen) {
+            *(frock->data) = xrealloc(*(frock->data), datalen);
+            memcpy(*(frock->data), data, datalen);
+        }
+        else {
+            /* datalen is 0: allocate a single byte and set it to "" */
+            *(frock->data) = xrealloc(*(frock->data), 1);
+            assert(data[0] == '\0');
```

Comment:
do we need this?  This whole block could just be `*frock->data = '\0';` or so.

Reply:
I've cleaned up the details a bit, but we do need the realloc juggle.  Even though the previous allocation is definitely long enough for one byte, the first time through there won't be a previous allocation.